lib/new_relic/harvest/collector/metric_data.ex

```diff
@@ -5,49 +5,55 @@ defmodule NewRelic.Harvest.Collector.MetricData do
 
-  def transform({:transaction, name}, duration_s: duration_s, total_time_s: total_time_s),
-    do: [
-      %Metric{
-        name: :HttpDispatcher,
-        call_count: 1,
-        total_call_time: duration_s,
-        total_exclusive_time: duration_s,
-        min_call_time: duration_s,
-        max_call_time: duration_s
-      },
-      %Metric{
-        name: :WebTransaction,
-        call_count: 1,
-        total_call_time: duration_s,
-        total_exclusive_time: duration_s,
-        min_call_time: duration_s,
-        max_call_time: duration_s
-      },
-      %Metric{
-        name: join(["WebTransaction", name]),
-        call_count: 1,
-        total_call_time: duration_s,
-        total_exclusive_time: duration_s,
-        min_call_time: duration_s,
-        max_call_time: duration_s
-      },
-      %Metric{
-        name: :WebTransactionTotalTime,
-        call_count: 1,
-        total_call_time: total_time_s,
-        total_exclusive_time: total_time_s,
-        min_call_time: total_time_s,
-        max_call_time: total_time_s
-      },
-      %Metric{
-        name: join(["WebTransactionTotalTime", name]),
-        call_count: 1,
-        total_call_time: total_time_s,
-        total_exclusive_time: total_time_s,
-        min_call_time: total_time_s,
-        max_call_time: total_time_s
-      }
-    ]
+  def transform({:transaction, name},
+        duration_s: duration_s,
+        total_time_s: total_time_s,
+        queue_duration_s: queue_duration_s
+      ),
+      do:
+        [
+          %Metric{
+            name: :HttpDispatcher,
+            call_count: 1,
+            total_call_time: duration_s,
+            total_exclusive_time: duration_s,
+            min_call_time: duration_s,
+            max_call_time: duration_s
+          },
+          %Metric{
+            name: :WebTransaction,
+            call_count: 1,
+            total_call_time: duration_s,
+            total_exclusive_time: duration_s,
+            min_call_time: duration_s,
+            max_call_time: duration_s
+          },
+          %Metric{
+            name: join(["WebTransaction", name]),
+            call_count: 1,
+            total_call_time: duration_s,
+            total_exclusive_time: duration_s,
+            min_call_time: duration_s,
+            max_call_time: duration_s
+          },
+          %Metric{
+            name: :WebTransactionTotalTime,
+            call_count: 1,
+            total_call_time: total_time_s,
+            total_exclusive_time: total_time_s,
+            min_call_time: total_time_s,
+            max_call_time: total_time_s
+          },
+          %Metric{
+            name: join(["WebTransactionTotalTime", name]),
+            call_count: 1,
+            total_call_time: total_time_s,
+            total_exclusive_time: total_time_s,
+            min_call_time: total_time_s,
+            max_call_time: total_time_s
+          }
+        ]
+        |> queue_duration_metric(queue_duration_s)
```

Comment:
we should report the queue metric separately from this, in this file we should just be adding a function like this:

```elixir
def transform(:queue_time, duration_s: duration_s) do
  [
    %Metric{}
  ]
end
```